pkg/csi/service/cns/controller.go

```diff
@@ -252,6 +269,39 @@ func (c *controller) DeleteVolume(ctx context.Context, req *csi.DeleteVolumeRequ
 	if err != nil {
 		return nil, err
 	}
+	// Get the list of persistent volumes from k8s
+	pvList, err := c.pvLister.List(labels.Everything())
+	if err != nil {
+		msg := fmt.Sprintf("DeleteVolume: Error getting PersistentVolumes from API server with err: %v", err)
+		klog.Error(msg)
+		return nil, status.Errorf(codes.Internal, msg)
+	}
+	var pvFound bool
+	var pvName string
+	// Find the PV name for the delete volume request
+	for _, pv := range pvList {
+		if pv.Spec.CSI != nil && pv.Spec.CSI.VolumeHandle == req.VolumeId {
+			pvFound = true
+			pvName = pv.Name
```

Comment:
Can we add some klog logs with log level 4 when PV is found, when it is attached, and when it is not attached to any node to help with debugging? make sure to print all references as we do not have trace id printed in this version of the driver.

Reply:
Done